legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/validation/FunctionParametersParametersValidation.java

```diff
@@ -67,12 +70,20 @@ private static void validateParameterValues(RichIterable<Variable> externalParam
         }
     }
 
-    public static ValidationResult validate(Variable var, Object value)
+    public static ValidationResult validate(Variable var, List<ValidationContext> validationContext, Object value)
     {
         FunctionParameterTypeValidator validator = FunctionParameterTypeValidator.externalParameterTypeValidator(var._class);
         if (validator == null)
         {
-            return ValidationResult.errorValidationResult("Unknown external parameter type: " + var._class + ", valid external parameter types: " + FunctionParameterTypeValidator.getExternalParameterTypes().makeString("[", ", ", "]"));
+            ValidationContext enumContext = validationContext.stream().filter(v -> (v.getClass() == EnumValidationContext.class)).filter(e -> ((EnumValidationContext) e).varName.equals(var.name)).findFirst().orElse(null);
```

Comment:
We should instanceOf rather than class equality.  On some circumstances where different class loaders re used, this could be false while instanceof will work regardless of class loader.

I also wonder if the visitor method added to the context abstract class could have help with encapsulating some of this logic.